demos/common/cpp/models/src/deblurring_model.cpp

```diff
@@ -14,63 +14,100 @@
+    }
+
+    if (inputShape.size() != 4 || inputShape[ov::layout::batch_idx(inputLayout)] != 1
+        || inputShape[ov::layout::channels_idx(inputLayout)] != 3) {
+        throw std::logic_error("3-channel 4-dimensional model's input is expected");
+    }
+
+    ov::preprocess::PrePostProcessor ppp(model);
+    ppp.input().tensor().
+        set_element_type(ov::element::u8).
+        set_layout("NHWC");
+
+    ppp.input().model().set_layout(inputLayout);
+
+    // --------------------------- Prepare output  -----------------------------------------------------
+    if (model->outputs().size() != 1) {
+        throw std::logic_error("Deblurring model wrapper supports topologies with only 1 output");
+    }
+
+    outputsNames.push_back(model->output().get_any_name());
+
+    const ov::Shape& outputShape = model->output().get_shape();
+    ov::Layout outputLayout("NCHW");
+    if (outputShape.size() != 4 || outputShape[ov::layout::batch_idx(outputLayout)] != 1
+        || outputShape[ov::layout::channels_idx(outputLayout)] != 3) {
+        throw std::logic_error("3-channel 4-dimensional model's output is expected");
+    }
+
+    ppp.output().tensor().set_element_type(ov::element::f32);
+    model = ppp.build();
+
+    changeInputSize(model);
 }
 
-void DeblurringModel::changeInputSize(InferenceEngine::CNNNetwork& cnnNetwork) {
-    InferenceEngine::ICNNNetwork::InputShapes inputShapes = cnnNetwork.getInputShapes();
-    InferenceEngine::SizeVector& inputDims = inputShapes.begin()->second;
+void DeblurringModel::changeInputSize(std::shared_ptr<ov::Model>& model) {
+    auto inTensorName = model->input().get_any_name();
+    ov::Shape inputShape = model->input().get_shape();
+    ov::Layout layout = ov::layout::get_layout(model->input());
 
-    if (inputDims[2] % stride || inputDims[3] % stride)
-        throw std::runtime_error("The shape of the model input must be divisible by stride");
+    auto batchId = ov::layout::batch_idx(layout);
+    auto heightId = ov::layout::height_idx(layout);
+    auto widthId = ov::layout::width_idx(layout);
+
+    if (inputShape[heightId] % stride || inputShape[widthId] % stride) {
+        throw std::logic_error("Model input shape HxW = "
+            + std::to_string(inputShape[heightId]) + "x" + std::to_string(inputShape[widthId])
+            + "must be divisible by stride " + std::to_string(stride));
+    }
 
     netInputHeight = static_cast<int>((netInputHeight + stride - 1) / stride) * stride;
     netInputWidth = static_cast<int>((netInputWidth + stride - 1) / stride) * stride;
 
-    inputDims[0] = 1;
-    inputDims[2] = netInputHeight;
-    inputDims[3] = netInputWidth;
+    inputShape[batchId] = 1;
+    inputShape[heightId] = netInputHeight;
+    inputShape[widthId] = netInputWidth;
 
-    cnnNetwork.reshape(inputShapes);
+    std::map<std::string, ov::PartialShape> shapes;
+    shapes[inTensorName] = ov::PartialShape(inputShape);
+    model->reshape(shapes);
```

Comment:
The reshaping in case of one input can be simplified like this:
```suggestion
    model->reshape({{inTensorName, inputShape}});
```

Reply:
It can't accept initializer llst, but instead we can just pass` inputShape.`

Reply:
I've launched with HPE demo with this change and it seems working well, I took it from [the cpp sample](https://github.com/openvinotoolkit/openvino/blob/master/samples/cpp/hello_reshape_ssd/main.cpp#L92).  
By the way, your modification is also suitable and even better.

Reply:
yes, you're right, I just missed the second pair of `{}`.